tiledb/api/examples/using_tiledb_stats.rs

```diff
@@ -0,0 +1,117 @@
+    let config: Config = tiledb::config::Config::new()?;
+    let vfs: VFS = tiledb::vfs::VFS::new(&tdb, &config)?;
+
+    let is_cur_dir = vfs.is_dir(ARRAY_NAME)?;
+    if is_cur_dir {
+        vfs.remove_dir(ARRAY_NAME)?;
+    }
+
+    let domain = {
+        let rows: tiledb::array::Dimension =
+            tiledb::array::DimensionBuilder::new::<u32>(
+                &tdb,
+                "row",
+                Datatype::UInt32,
+                &[1, 12000],
+                &row_tile_extent,
+            )?
+            .build();
+        let cols: tiledb::array::Dimension =
+            tiledb::array::DimensionBuilder::new::<u32>(
+                &tdb,
+                "col",
+                Datatype::UInt32,
+                &[1, 12000],
+                &col_tile_extent,
+            )?
+            .build();
+
+        tiledb::array::DomainBuilder::new(&tdb)?
+            .add_dimension(rows)?
+            .add_dimension(cols)?
+            .build()
+    };
+
+    let attribute_a = tiledb::array::AttributeBuilder::new(
+        &tdb,
+        ATTRIBUTE_NAME,
+        tiledb::Datatype::Int32,
+    )?
+    .build();
+
+    let schema = tiledb::array::SchemaBuilder::new(
+        &tdb,
+        tiledb::array::ArrayType::Dense,
+        domain,
+    )?
+    .add_attribute(attribute_a)?
+    .build();
+
+    tiledb::Array::create(&tdb, ARRAY_NAME, schema)
+}
+
+pub fn write_array() -> TileDBResult<()> {
+    let tdb = tiledb::context::Context::new()?;
+    let array: Array = tiledb::Array::open(&tdb, ARRAY_NAME, tiledb::array::Mode::Write)?;
+    let mut data : Vec<i32> =  Vec::from_iter(0..12000 * 12000);
+
+    let query =
+        tiledb::QueryBuilder::new(&tdb, array, tiledb::QueryType::Write)?
+            .layout(tiledb::array::Layout::RowMajor)?
+            .dimension_buffer_typed(
+                ATTRIBUTE_NAME,
+                data.as_mut_slice(),
+            )?
+            .build();
+
+    query.submit().map(|_| ())
+}
+
+pub fn read_array() -> TileDBResult<()> {
+    let tdb = tiledb::context::Context::new()?;
+
+    let array = tiledb::Array::open(
+        &tdb,
+        ARRAY_NAME,
+        tiledb::array::Mode::Read,
+    )?;
+
+    let mut results = vec![0; 3000 * 12000];
+
+    let query =
+        tiledb::QueryBuilder::new(&tdb, array, tiledb::QueryType::Read)?
+            .layout(tiledb::array::Layout::RowMajor)?
+            .dimension_buffer_typed(
+                ARRAY_NAME,
+                results.as_mut_slice(),
+            )?
+            .add_subarray()?
+            .dimension_range_typed::<i32>(0, &[1, 3000])?
+            .add_subarray()?
+            .dimension_range_typed::<i32>(1, &[1, 12000])?
+            .build();
+
+    tiledb::stats::stats_enable()?;
+    query.submit()?;
+    let stats : String = tiledb::stats::stats_dump_str()?;
+    println!("{}", &stats);
+    tiledb::stats::stats_disable().map(|_| ())
+}
+
+fn main() {
+    let _ = create_array(1, 12000);
+    let _ = write_array();
```

Comment:
This suppresses the warning but the warning is there for a good reason.  I think an `expect` like in `quickstart_dense` is fine but there does need to be a notification if anything fails.

Reply:
My vague style preference is to only expect things that might fail that are being tested, everything else that we're not testing should be `?` to indicate its super unexpected to fail.

Reply:
One of the unsatisfying aspects of the `Result` and `?` operators is that you do miss a lot of context when you finally get to the end.  I generally agree that `expect` should be avoided inside a call stack, but `main` is at the bottom, there's nothing else left to catch and handle the error except for the system which can't possibly add any context.  I endorse `expect` in the `main` function because then if it does fail, then there is a clean message on the command line saying where to start, which you might not get from the default "`main` returned `Err`" handler.